### ISSUE TITLE
[DateTimePicker] Support masked input for AM/PM locales

### ISSUE THREAD

**Issue by siropkin** [mui/mui-x#4869]
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

DateTimePicker ignoring `mask` prop if  `ampm == true` or `ampm unset`.

## Expected Behavior 🤔

DateTimePicker should not ignoring `mask` prop if  `ampm == true` or `ampm unset`.

## Steps to Reproduce 🕹
If I will put `mask` prop and set `ampm` prop to `true` then DateTimePicker will ignore `mask` prop.

Quick code example:
```
const inputFormat = "yyyy-MM-dd hh:mm:ss";
const mask = "____-__-__ __:__:__";
...
<DateTimePicker
	renderInput={(props) => <TextField {...props} />}
	label="DateTimePicker"
	value={value}
	onChange={(newValue) => {
	  setValue(newValue);
	}}
	inputFormat={inputFormat}
	mask={mask}
	ampm
/>
```

Full code example: https://codesandbox.io/s/basicdatetimepicker-material-demo-forked-zkl5y?file=/demo.js

Workaround is to set `disableMaskedInput` to `false`:
```
const inputFormat = "yyyy-MM-dd hh:mm:ss";
const mask = "____-__-__ __:__:__";
...
<DateTimePicker
	renderInput={(props) => <TextField {...props} />}
	label="DateTimePicker"
	value={value}
	onChange={(newValue) => {
	  setValue(newValue);
	}}
	inputFormat={inputFormat}
	mask={mask}
	ampm
        disableMaskedInput={false}
/>
```

## Your Environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>
  
```
  System:
    OS: Windows 10 10.0.19042
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 3.81 GB / 15.82 GB
  Binaries:
    Node: 12.22.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.10 - C:\Program Files\nodejs\yarn.CMD
    npm: 6.14.12 - C:\Program Files\nodejs\npm.CMD
  Managers:
    pip3: 21.0.1 - C:\Python39\Scripts\pip3.EXE
  Utilities:
    Git: 2.33.0. - /mingw64/bin/git
  Virtualization:
    Docker: 20.10.8 - C:\Program Files\Docker\Docker\resources\bin\docker.EXE
  Languages:
    Bash: 4.4.23 - C:\Program Files\Git\usr\bin\bash.EXE
    Perl: 5.34.0 - C:\Program Files\Git\usr\bin\perl.EXE
    Python: 3.9.2 - /c/Python39/python
  Browsers:
    Chrome: 92.0.4515.159
    Edge: Spartan (44.19041.1023.0), Chromium (93.0.961.38)
```
</details>

**Comment by flaviendelangle** [comment]
@alexfauquette I let you have a look at this one :+1:

**Comment by alexfauquette** [comment]
This has been fixed in #4731

Available in `v5.0.0-alpha.3`
